instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java

```diff
@@ -58,13 +59,18 @@
   public static <REQUEST, RESPONSE> InstrumenterBuilder<REQUEST, RESPONSE> newBuilder(
       OpenTelemetry openTelemetry,
       String instrumentationName,
+      InstrumentationType instrumentationType,
       SpanNameExtractor<? super REQUEST> spanNameExtractor) {
-    return new InstrumenterBuilder<>(openTelemetry, instrumentationName, spanNameExtractor);
+    return new InstrumenterBuilder<>(openTelemetry, instrumentationName, instrumentationType, spanNameExtractor);
```

Comment:
Since `instrumentationType` is only for client spans(?), it would be nice to scope it to just client instrumenters. Let's see if @anuraaga and @mateuszrzeszutek have any thoughts how to do this.

Reply:
I recall we thought it would be useful to have a server span key as well, but as a way to enrich it rather than suppress it. Is there a need in java, e.g. in Spring to ever suppress server spans? I remember such cases in .NET world.  
Anyway, agree that beyond client spans it's not that important

Reply:
oh that makes sense, we do suppress nested SERVER spans in Java instrumentation today. i c now, since `ClientSpan` creates it's own context keys, those wouldn't overlap with context keys created by `ServerSpan` 👍